File: Lua/comb_sort.lua
function comb_sort(arr)
    local gap = #arr
    local shrink = 1.3
    local swapped = true

    while gap > 1 or swapped do
        gap = math.floor(gap / shrink)
        if gap < 1 then gap = 1 end
        swapped = false

        for i = 1, #arr - gap do
            if arr[i] > arr[i + gap] then
                arr[i], arr[i + gap] = arr[i + gap], arr[i]
                swapped = true
            end
        end
    end
end

local arr = {5, 1, 4, 2, 8}
comb_sort(arr)
print(table.concat(arr, " "))

File: JavaScript/comb_sort.js
function combSort(arr) {
    const shrinkFactor = 1.3;
    let gap = arr.length;
    let sorted = false;

    while (!sorted) {
        gap = Math.floor(gap / shrinkFactor);
        if (gap <= 1) {
            gap = 1;
            sorted = true;
        }

        for (let i = 0; i + gap < arr.length; i++) {
            if (arr[i] > arr[i + gap]) {
                [arr[i], arr[i + gap]] = [arr[i + gap], arr[i]];
                sorted = false;
            }
        }
    }

    return arr;
}

const array = [64, 34, 25, 12, 22, 11, 90];
console.log(combSort(array));